stdlib/src/collections/optional.mojo

```diff
@@ -154,6 +154,56 @@ struct Optional[T: CollectionElement](CollectionElement, Boolable):
         """
         return not self
 
+    fn __str__[U: RepresentableCollectionElement](self: Optional[U]) -> String:
+        """Return the string representation of the value of the Optional.
+
+        Parameters:
+            U: The type of the elements in the list. Must implement the
+              traits `Representable` and `CollectionElement`.
+
+        Returns:
+            A string represenation of the Optional.
+        """
+        var output = String()
+        var writer = output._unsafe_to_formatter()
+        self.format_to(writer)
+        return output
+
+    # TODO: Include the Parameter type in the string as well.
+    fn __repr__[U: RepresentableCollectionElement](self: Optional[U]) -> String:
+        """Returns the verbose string representation of the Optional.
+
+        Parameters:
+            U: The type of the elements in the list. Must implement the
+              traits `Representable` and `CollectionElement`.
+
+        Returns:
+            A verbose string representation of the Optional.
+        """
+        var output = String()
+        var writer = output._unsafe_to_formatter()
+        write_to(writer, "Optional(")
+        self.format_to(writer)
+        write_to(writer, ")")
+        return output
+
+    fn format_to[
+        U: RepresentableCollectionElement, //
+    ](self: Optional[U], inout writer: Formatter):
+        """Write Optional string representation to a `Formatter`.
+
+        Parameters:
+            U: The type of the elements in the list. Must implement the
+              traits `Representable` and `CollectionElement`.
+
+        Args:
+            writer: The formatter to write to.
+        """
+        if self:
+            write_to(writer, repr(self.value()))
```

Comment:
Here `repr(self.value())` can be a very big `String`. If we know that `self.value()` has a `format_to` method, would it be possible to use it to write directly to the `Formatter`?

You can imagine that `self.value()` is a List with 10 000 elements for example.

Reply:
I like the idea, but wouldn't that require use to generally shift away from `RepresentableCollectionElement` in these scenarios to something equivalent to `Formattable & CollectionElement`?

Reply:
I agree with Gabriel that we could make this more efficient, though we can do that in a follow-up PR if you'd like 🙂

Reply:
Might as well do it here. I'm on vacation at the moment but I can look into it on Monday!